src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -72,17 +72,14 @@ def validate_request_update_args(request_args, config, reqmgr_db_service, param)
     
     TODO: rasie right kind of error with clear message 
     """
-
-    request_name = request_args["RequestName"]
-    # this need to be deleted for validation
-    del request_args["RequestName"]
+    # this needs to be deleted for validation
+    request_name = request_args.pop("RequestName")
```

Comment:
The pop method will throw exception if key does not exists, therefore it's better to be safe and but try/except KeyError around it. But try/except is costly, and it's better just to get request name and then delete it.